subsys/bluetooth/mesh/gatt_cli.c

```diff
@@ -0,0 +1,349 @@
+		BT_WARN("Too small Proxy PDU");
+		return BT_GATT_ITER_STOP;
+	}
+
+	(void)bt_mesh_proxy_msg_recv(conn, val, length);
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static uint8_t discover_func(struct bt_conn *conn,
+			     const struct bt_gatt_attr *attr,
+			     struct bt_gatt_discover_params *params)
+{
+	int err;
+	struct bt_mesh_gatt_server *server;
+
+	if (!attr) {
+		BT_DBG("GATT Services Discover complete");
+		(void)memset(params, 0, sizeof(*params));
+		return BT_GATT_ITER_STOP;
+	}
+
+	BT_DBG("[ATTRIBUTE] handle %u", attr->handle);
+
+	server = find_server(conn);
+
+	if (!bt_uuid_cmp(server->discover.uuid, &server->gatt->srv_uuid.uuid)) {
+		server->svc_start_handle = attr->handle;
+
+		server->discover.uuid = &server->gatt->data_in_uuid.uuid;
+		server->discover.start_handle = attr->handle + 1;
+		server->discover.type = BT_GATT_DISCOVER_CHARACTERISTIC;
+
+		err = bt_gatt_discover(conn, &server->discover);
+		if (err) {
+			BT_DBG("Discover GATT data in char failed (err %d)", err);
+		}
+	} else if (!bt_uuid_cmp(server->discover.uuid,
+				&server->gatt->data_in_uuid.uuid)) {
+		server->data_in_handle = attr->handle + 1;
+
+		server->discover.uuid = &server->gatt->data_out_uuid.uuid;
+		server->discover.start_handle = server->svc_start_handle + 1;
+		server->discover.type = BT_GATT_DISCOVER_CHARACTERISTIC;
+
+		err = bt_gatt_discover(conn, &server->discover);
+		if (err) {
+			BT_DBG("Discover GATT data out char failed (err %d)", err);
+		}
+	} else if (!bt_uuid_cmp(server->discover.uuid,
+				&server->gatt->data_out_uuid.uuid)) {
+		server->discover.uuid = &server->gatt->data_out_cccd_uuid.uuid;
+		server->discover.start_handle = attr->handle + 2;
+		server->discover.type = BT_GATT_DISCOVER_DESCRIPTOR;
+
+		err = bt_gatt_discover(conn, &server->discover);
+		if (err) {
+			BT_DBG("Discover GATT CCCD failed (err %d)", err);
+		}
+	} else {
+		server->subscribe.notify = notify_func;
+		server->subscribe.value = BT_GATT_CCC_NOTIFY;
+		server->subscribe.ccc_handle = attr->handle;
+		server->subscribe.value_handle = attr->handle - 1;
+
+		err = bt_gatt_subscribe(conn, &server->subscribe);
+		if (err && err != -EALREADY) {
+			BT_DBG("Subscribe failed (err %d)", err);
+		} else {
+			BT_DBG("[SUBSCRIBED]");
+
+			server->gatt->link_open(conn);
+		}
+	}
+
+	return BT_GATT_ITER_STOP;
+}
+
+int bt_mesh_gatt_send(struct bt_conn *conn,
+		      const void *data, uint16_t len,
+		      bt_gatt_complete_func_t end, void *user_data)
+{
+	struct bt_mesh_gatt_server *server;
+
+	server = find_server(conn);
+
+	BT_DBG("%u bytes: %s", len, bt_hex(data, len));
+
+	return bt_gatt_write_without_response_cb(conn, server->data_in_handle,
+						 data, len, false, end, user_data);
+}
+
+static void gatt_connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	struct bt_mesh_gatt_server *server;
+	struct bt_conn_info info;
+	int err;
+
+	bt_conn_get_info(conn, &info);
+	if (info.role != BT_CONN_ROLE_CENTRAL) {
+		return;
+	}
+
+	server = find_server(conn);
```

Comment:
I'd rename this "get" instead of "find" (since it can't fail) and move it up to the variable declaration, so that the function starts with:
```c
{
	struct bt_messh_gatt_server *server = get_server(conn);
```